src/ansiblelint/rules/var_naming.py

```diff
@@ -360,6 +396,28 @@ def test_var_naming_with_role_prefix(
         for result in results:
             assert result.tag == "var-naming[no-role-prefix]"
 
+    def test_var_naming_with_role_prefix_plays(
+        default_rules_collection: RulesCollection,
+    ) -> None:
+        """Test rule matches."""
+        results = Runner(
+            Lintable("examples/playbooks/role_vars_prefix_detection.yml"),
+            rules=default_rules_collection,
+            exclude_paths=["examples/roles/role_vars_prefix_detection"],
```

Comment:
`exclude_paths` was set because otherwise the runner picks up the violations from the role too, I presume this happens because the playbook invokes said role.